client/blocks/video-editor/index.jsx

```diff
@@ -0,0 +1,230 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { noop } from 'lodash';
+import { localize } from 'i18n-calypso';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import ProgressBar from 'components/progress-bar';
+import Notice from 'components/notice';
+import DetailPreviewVideo from 'post-editor/media-modal/detail/detail-preview-video';
+import VideoEditorButtons from './video-editor-buttons';
+import { updatePoster } from 'state/ui/editor/video-editor/actions';
+import {
+	getPosterUploadProgress,
+	getPosterUrl,
+	shouldCloseVideoEditorModal,
+	shouldShowVideoEditorError,
+} from 'state/selectors';
+
+/**
+ * Module variables
+ */
+let isSelectingFrame = false;
+
+class VideoEditor extends Component {
+	static propTypes = {
+		className: PropTypes.string,
+		media: PropTypes.object.isRequired,
```

Comment:
Also optional, but the component may be easier to use if we can handle media not being available initially.